script.cinder/resources/lib/CinderRandomPlayer.py

```diff
@@ -0,0 +1,312 @@
+    def bootstrapPlayback(self):
+        skipFirstSource = True
+
+        indexList = range(0, len(self.configSettings['SourceUriList']))
+        if self.configSettings['SkipSources'] == "true" or \
+           self.configSettings['ShufflePlaylist'] == "true":
+            skipFirstSource = False
+            random.shuffle(indexList)
+            random.shuffle(indexList)
+            random.shuffle(indexList)
+
+        # attempt to find an episode from any sources only once 
+        # let the caller do multiple sweeps
+        for index in indexList: 
+            showWeight = int(self.configSettings['SourceWeightList'][index])
+            if random.randint(1, 100) > showWeight:
+                continue
+
+            videos = self.getRandomEpisode(self.configSettings['SourceUriList'][index])
+
+            # skip duplicate videos
+            if self.isDuplicate(videos): 
+                # should not hit this since its the first episode queued up
+                # need to call isDuplicate to have it added to in internal
+                # list
+                continue
+
+            self.playRandomPlaylist(videos)
+            return skipFirstSource
+        return skipFirstSource
+
+
+    # Returns True if the episode has already been queued up. Keeps track of which
+    # episodes have been provided as a parameter
+    def isDuplicate(self, randomFileList):
+        # there should only be one file in randomFileList
+        for randomFile in randomFileList:
+            for queuedEpisodeDuplicate in self.queuedEpisodesDuplicateCheck:
+                if randomFile['file'] == queuedEpisodeDuplicate['file']: 
+                    return True
+            self.queuedEpisodesDuplicateCheck.append(randomFile)
+            return False
+        return False
+
+
+    # Plays random episodes according to the user specified configuration settings
+    def playRandomEpisodes(self):
+        xbmc.executebuiltin('ActivateWindow(busydialog)')
+
+        self.queuedEpisodesDuplicateCheck = []
+        self.queuedEpisodes = None
+        numberOfShows = int(self.configSettings['NumberOfShows']) 
+
+        skipFirstSource = False
+        if self.configSettings['BootstrapPlayback'] == "true":
+            skipFirstSource = self.bootstrapPlayback()
+            numberOfShows -= 1
+
+            # catch the case of the user only wanting 1 episode queued up
+            if numberOfShows <= 0:
+                return
+
+        episodeList = []
+        sweepCount = 0
+        while True:
+
+            sweepCount += 1
+            if sweepCount > 100:
+                # after 100 attempts to fill up the playlist cancel the process
+                # the user is probably in 'watched' or 'unwatched' only modes
+                # and the process is not finding anything
+                xbmcgui.Dialog().notification('Cinder Error:', 'Unable to find episodes. Try "All videos" mode', xbmcgui.NOTIFICATION_WARNING)
```

Comment:
if it´s only error messages etc then I´d say it´s fine for a start. Most scripts never end up on transifex anyways....